mathics/builtin/calculus.py

```diff
@@ -273,9 +348,12 @@ class Derivative(PostfixOperator, SympyFunction):
             r'    RowBox[{"(", Sequence @@ Riffle[{n}, ","], ")"}]]]]'
         ),
         "MakeBoxes[Derivative[n:1|2][f_], form:OutputForm]": """RowBox[{MakeBoxes[f, form], If[n==1, "'", "''"]}]""",
+        # The following rules should be applied in the apply method, instead of relying on the pattern matching
+        # mechanism.
         "Derivative[0...][f_]": "f",
         "Derivative[n__Integer][Derivative[m__Integer][f_]] /; Length[{m}] "
         "== Length[{n}]": "Derivative[Sequence @@ ({n} + {m})][f]",
+        # This would require at least some comments...
```

Comment:
Ok - I guess this isn't finished? Or haven't decided what to do?

Reply:
Here I am not sure what to do. As it is, it works with the functions I imagined it makes sense to try, but I would like to have something clearer here. Also, I would like to avoid relying on the pattern matching mechanism when it could be handled by a more or less concise code.

Reply:
By now, and to avoid making this PR too fat, I would merge this as it is, and then continue making improvementsments in another PR.